libs/datamodel/core/tests/types/negative.rs

```diff
@@ -109,6 +109,29 @@ fn should_fail_on_custom_related_types() {
     ));
 }
 
+#[test]
+fn should_fail_on_native_type_with_invalid_datasource_name() {
+    let dml = r#"
+        datasource db {
+          provider = "postgres"
+          url = "postgresql://"
+          previewFeatures = ["nativeTypes"]
+        }
+
+        model Blog {
+            id     Int    @id
+            bigInt Int    @pg.BigInt
+        }
+    "#;
+
+    let error = parse_error(dml);
+
+    error.assert_is(DatamodelError::new_connector_error(
+        "Native types prefix pg is not compatible with the datasource used in this schema, expected prefix db.",
```

Comment:
That error message is indeed correct but i think it might be a bit hard to understand for users. We should make it a bit more explicit imo. Maybe something like this? WDYT?
> The prefix `pg` is invalid. It must equal the name of an existing datasource e.g. `db`. Did you mean to use `@db.BigInt`?

Reply:
Yes, that make it much more clear 👍